src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -121,8 +124,9 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 				this._decorations.set(decoration.marker.id, { decoration, disposables });
 			}
 			if (decoration.element?.clientWidth! > 0) {
-				const marginWidth = ((decoration.element?.parentElement?.parentElement?.previousElementSibling?.clientWidth || 0) - (decoration.element?.parentElement?.parentElement?.clientWidth || 0)) * .5;
-				target.style.marginLeft = `${((marginWidth - (decoration.element!.clientWidth + DecorationStyles.ButtonMargin)) * .5) - marginWidth}px`;
+				// if this is the leftmost terminal, there is extra room
+				const marginWidth = document.querySelectorAll(`.${DecorationSelector.XtermScreen}`)[0] === decoration.element?.parentElement?.parentElement ? 15 : 10;
+				target.style.marginLeft = `${-marginWidth - 2}px`;
```

Comment:
This would be better off in CSS near the other rules around splits like

https://github.com/microsoft/vscode/blob/2ac67c722c98d768c2260ec5ff0fa9a8ce4d2848/src/vs/workbench/contrib/terminal/browser/media/terminal.css#L96-L103